### PR TITLE
test: improve test coverage for apps/api orgs routes

### PR DESCRIPTION
**Pull request by is0692vs**
Adds missing unit test coverage for apps/api/src/routes/orgs.ts route logic, including permissions blocks, data edge cases, and unexpected DB errors.

---
*PR created automatically by Jules for task [9203649208671370512](https://jules.google.com/task/9203649208671370512) started by @is0692vs*

<details open><summary><h3>Greptile Summary</h3></summary>

このPRは `apps/api/src/routes/orgs.ts` に対するテストカバレッジを大幅に拡充しており、バリデーション失敗・権限チェック・DB エラー（非 UNIQUE 制約）など多くの未カバーブランチにテストを追加しています。全体的な構成は正しく、モックのセットアップも概ね適切です。
</details>

<details open><summary><h3>Confidence Score: 5/5</h3></summary>

本PRはテストファイルのみの変更であり、プロダクションコードに影響はなく安全にマージできます。

P1/P0 の問題は見つかりませんでした。P2 の指摘（アサーションの弱さ・mockDb 直接上書きの不統一）があるものの、いずれも非ブロッキングな改善提案です。

特に注意が必要なファイルはありません。
</details>

<details open><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/api/src/routes/__tests__/orgs.test.ts | org ルートのテストカバレッジを大幅拡充（バリデーション・権限・DB エラー系）。モックのセットアップが一部不統一で、可視性フィルタリングテストのアサーションが弱い。 |

</details>

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[POST /api/orgs] --> A1[バリデーション: description, slug, name]
    A1 -->|エラー| A2[400]
    A1 -->|OK| A3[slug 重複チェック]
    A3 -->|重複| A4[409]
    A3 -->|OK| A5[INSERT orgs + orgMembers]
    A5 -->|DB エラー UNIQUE| A4
    A5 -->|DB エラー その他| A6[500 ★新テスト]
    A5 -->|成功| A7[201]

    B[PATCH /api/orgs/:slug] --> B1[getOrgBySlug]
    B1 -->|null| B2[404 ★未テスト]
    B1 -->|OK| B3[requireOrgAdmin]
    B3 -->|失敗| B4[403]
    B3 -->|OK| B5[バリデーション ★新テスト]
    B5 -->|エラー| B6[400]
    B5 -->|OK| B7[UPDATE]

    C[POST /api/orgs/:slug/members] --> C1[getOrgBySlug + requireOrgAdmin]
    C1 --> C2[バリデーション ★新テスト]
    C2 -->|userId/role エラー| C3[400]
    C2 -->|OK| C4[ユーザー存在確認 ★新テスト]
    C4 -->|null| C5[404]
    C4 -->|OK| C6[INSERT orgMembers]
    C6 -->|UNIQUE| C7[409]
    C6 -->|その他 DB エラー| C8[500 ★新テスト]
```

<details open><summary><h3>Comments Outside Diff (3)</h3></summary>

1. `apps/api/src/routes/__tests__/orgs.test.ts`, line 253-274 ([link](https://github.com/hiroki-org/openshelf/blob/ecb00425119cc97787015c808780b5d113218167/apps/api/src/routes/__tests__/orgs.test.ts#L253-L274)) 

   <a href="#">P2</a> **重複するテストケース**

   `"returns 400 for invalid JSON body"` (163行目) と `"returns 400 for invalid JSON bodies"` (253行目) の2つのテストは、同じコードパス（`POST /api/orgs` のJSON解析失敗）を異なる無効な入力値（`"invalid json string"` vs `"{"`)でテストしているだけで、実質的に重複しています。どちらか一方を削除するか、またはより意味のある差別化（例: 境界値）が必要です。

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: apps/api/src/routes/__tests__/orgs.test.ts
   Line: 253-274

   Comment:
   **重複するテストケース**

   `"returns 400 for invalid JSON body"` (163行目) と `"returns 400 for invalid JSON bodies"` (253行目) の2つのテストは、同じコードパス（`POST /api/orgs` のJSON解析失敗）を異なる無効な入力値（`"invalid json string"` vs `"{"`)でテストしているだけで、実質的に重複しています。どちらか一方を削除するか、またはより意味のある差別化（例: 境界値）が必要です。

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>

2. `apps/api/src/routes/__tests__/orgs.test.ts`, line 897-924 ([link](https://github.com/hiroki-org/openshelf/blob/ecb00425119cc97787015c808780b5d113218167/apps/api/src/routes/__tests__/orgs.test.ts#L897-L924)) 

   <a href="#">P2</a> **対象ユーザーのロールが誤ってモックされている**

   `selectCallCount >= 2` 以降は全て `{ role: "admin" }` を返しているため、select #3（`getOrgMembership` でターゲットユーザーのメンバーシップ取得）も `"admin"` を返します。今回は `"superadmin"` というロールが渡されるためバリデーションで先に 400 が返り偶然通過しますが、モックのセットアップが意図を正確に表していません。ターゲットの membership に `{ role: "member" }` を明示的に返すよう修正することを検討してください。

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: apps/api/src/routes/__tests__/orgs.test.ts
   Line: 897-924

   Comment:
   **対象ユーザーのロールが誤ってモックされている**

   `selectCallCount >= 2` 以降は全て `{ role: "admin" }` を返しているため、select #3（`getOrgMembership` でターゲットユーザーのメンバーシップ取得）も `"admin"` を返します。今回は `"superadmin"` というロールが渡されるためバリデーションで先に 400 が返り偶然通過しますが、モックのセットアップが意図を正確に表していません。ターゲットの membership に `{ role: "member" }` を明示的に返すよう修正することを検討してください。

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>

3. `apps/api/src/routes/__tests__/orgs.test.ts`, line 319-521 ([link](https://github.com/hiroki-org/openshelf/blob/ecb00425119cc97787015c808780b5d113218167/apps/api/src/routes/__tests__/orgs.test.ts#L319-L521)) 

   <a href="#">P2</a> **`PATCH /api/orgs/:slug` に org-not-found のテストケースがない**

   `PATCH /api/orgs/:slug` のルートは `getOrgBySlug` が null を返した場合に 404 を返しますが（`orgs.ts` 338行目）、このテストスイートにはその分岐をカバーするテストケースが存在しません。同様に `DELETE /api/orgs/:slug` の 404 ケースも未テストです。

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: apps/api/src/routes/__tests__/orgs.test.ts
   Line: 319-521

   Comment:
   **`PATCH /api/orgs/:slug` に org-not-found のテストケースがない**

   `PATCH /api/orgs/:slug` のルートは `getOrgBySlug` が null を返した場合に 404 を返しますが（`orgs.ts` 338行目）、このテストスイートにはその分岐をカバーするテストケースが存在しません。同様に `DELETE /api/orgs/:slug` の 404 ケースも未テストです。

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>
</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
Fix the following 2 code review issues. Work through them one at a time, proposing concise fixes.

---

### Issue 1 of 2
apps/api/src/routes/__tests__/orgs.test.ts:642-671
**可視性フィルタリングのアサーションが弱い**

テスト名「returns properly filtered papers for non-members who are authors」はメンバーでない著者に対する可視性フィルタリングを検証するものですが、`expect(body.papers).toBeDefined()` しか確認していません。DB がモックされているため実際のフィルタ条件は適用されず、`paper-public`・`paper-org`・`paper-private` の各論文が正しく含まれているかどうかが検証されていません。テスト名に即した意味のあるアサーションを追加してください。

```suggestion
            expect(res.status).toBe(200);
            const body = (await res.json()) as any;
            expect(body.papers).toBeDefined();
            // 著者であるため org_only・private の論文も含まれていることを確認
            expect(body.papers.map((p: any) => p.id).sort()).toEqual(
                ["paper-org", "paper-private", "paper-public"].sort(),
            );
```

### Issue 2 of 2
apps/api/src/routes/__tests__/orgs.test.ts:58-89
**`queueSelectResponses` を使わず直接 `mockDb.select` を書き換えている**

このテストだけ `mockDb.select = vi.fn(() => makeQuery({ getResult: null }))` と直接上書きしていますが、ファイル内の他のテストはすべて `queueSelectResponses` を使っています。同じヘルパーを使って記述すると一貫性が向上します。

```suggestion
            queueSelectResponses([{ getResult: null }]);
```


`````

</details>

<sub>Reviews (2): Last reviewed commit: ["test(api): finalize orgs route coverage ..."](https://github.com/hiroki-org/openshelf/commit/20cdb31d4fa922092a0125aabde116137229b70b) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=30339151)</sub>

### CODE REVIEW

**Comment by is0692vs** [comment]
対応しました: インデントを修正しました。影響範囲: apps/api/src/routes/__tests__/orgs.test.ts の app.request() 呼び出しのインデント。 検証: linterとテストが通ることを確認。

File: apps/api/src/routes/__tests__/orgs.test.ts [hunk 1]
```diff
@@ -33,6 +33,61 @@ describe("orgs routes", () => {
...
+                    {
+                        method: "POST",
+                        headers: {
+                            Authorization: `Bearer ${token}`,
+                            "Content-Type": "application/json",
+                        },
+                        body: JSON.stringify({ name: "Lab", slug: "lab" }),
+                    },
+                    env as any,
+                );
```
[the remark this answers is not in the document]
Comment (is0692vs) [reply apps/api/src/routes/__tests__/orgs.test.ts]: 対応しました: app.request 呼び出しのインデントを周囲のテストと同じレベルに揃えました。検証: npm run typecheck --workspace api / npm run test --workspace api -- src/routes/__tests__/orgs.test.ts.

**Comment by is0692vs** [comment]
@greptile
